### ISSUE TITLE
Recount command works bad

### ISSUE THREAD

**Issue by VozikCZE** [BentoBoxWorld/Limits#123]
**Description**
Command for recount limit works only sometimes.

**Steps to reproduce the behavior:**
1. Use /bsb limits calc or /is limits recount
2. Wait.
Nothing will usually happen, it only sometimes recount the level - I trying to find when it recount, but I didin't find any reason yet.

**Expected behavior**
It should recount limits with some messages in chat like "Recounting island level... Estimated time: 50 seconds"

**Server Information:**

[Please complete the following information:]
[09:55:16 INFO]: Běžím na Neplatné (PURPUR) Neplatné.
[09:55:16 INFO]: Verze BentoBox: 1.15.5
[09:55:16 INFO]: Databáze: MYSQL
[09:55:16 INFO]: Načtené herní světy:
[09:55:16 INFO]: bskyblock_world (BSkyBlock): Svět, Nether, End
[09:55:16 INFO]: Načtené doplňky:
[09:55:16 INFO]: BSkyBlock 1.14.3 (ENABLED)
[09:55:16 INFO]: Chat 1.1.2 (ENABLED)
[09:55:16 INFO]: EpicHooks 1.1.0-SNAPSHOT-b5 (ENABLED)
[09:55:16 INFO]: Greenhouses 1.5.1-SNAPSHOT-b287 (ENABLED)
[09:55:16 INFO]: Level 2.5.0 (ENABLED)
[09:55:16 INFO]: Limits 1.15.0 (ENABLED)
[09:55:16 INFO]: Warps 1.9.7 (ENABLED)

**Comment by ChimpGamer** [comment]
If I check /is top the amount of hoppers is not the same with /is limits even after executing /is limits recount.
![image](https://user-images.githubusercontent.com/19960733/116883863-5d5c5900-ac26-11eb-89ff-f2eed4be9ee9.png)
![image](https://user-images.githubusercontent.com/19960733/116883929-6fd69280-ac26-11eb-9599-05f91e43c84e.png)

**Comment by VozikCZE** [comment]
Limits have more problems with bad counting.

**Comment by tastybento** [comment]
Yes. It needs reworking to essentially use the same code as Level. It's on the list.

**Comment by VozikCZE** [comment]
@tastybento I think this is important issue and it should be fixed soon.. some players are stuck because they have wrong count in /is limits and they can't recount it.

**Comment by tastybento** [comment]
Okay. I'll see what I can do.

**Comment by tastybento** [comment]
This update should fix the block recounting. It does not include entity recounting yet. Note that it recounts for overworld, nether, and end, if they are enabled.

**Comment by tastybento** [comment]
@VozikCZE Okay, please try the snapshot and let me know if it works for you.

**Comment by VozikCZE** [comment]
@tastybento Sorry for the delay.. It works slowly, but works :D Also it doesn't have any spam protection.

**Comment by tastybento** [comment]
**Assessment (July 2026):** Closing as fixed. The recount was rewritten after this report (chunk-snapshot-based scan, per-environment, run through BentoBox's pipeline), and the reporter confirmed it working. Since then the counting subsystem has had a further overhaul — counts are persistent per environment and the recount rebuilds them wholesale. The "no spam protection" note is covered by the player-command cooldown (`cooldown: 120` in config).

If a recount still produces a wrong number on a current build, that's a new bug — please open a fresh issue with the material, the expected vs. reported count, and whether nether/end are enabled.
